File: app/src/main/java/com/wireguard/util/Keyed.java
/*
 * Copyright © 2017-2019 WireGuard LLC. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

package com.wireguard.util;

/**
 * Interface for objects that have a identifying key of the given type.
 */

public interface Keyed<K> {
    K getKey();
}

File: ui/src/main/java/com/hezwin/android/databinding/Keyed.kt
/*
 * Copyright © 2017-2025 HEZWIN LLC. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */
package com.hezwin.android.databinding

/**
 * Interface for objects that have a identifying key of the given type.
 */
interface Keyed<K> {
    val key: K
}
